lib/Core/Site/RelationService.php

```diff
@@ -107,7 +110,7 @@ private function getRelatedContentItems(array $relatedContentIds, array $content
 
         $query = new Query([
             'filter' => $criteria,
-            'limit' => count($relatedContentIds),
+            'limit' => $limit ? $limit : count($relatedContentIds),
```

Comment:
This should probably be cast to `int` and check if larger than zero before using it.

Reply:
Thanks. Fixed in [01ae43](https://github.com/netgen/ezplatform-site-api/pull/68/commits/01ae43193b234df92a60bfa5c301525934651584)